examples/llm/main.py

```diff
@@ -64,6 +69,23 @@ def main(cfg):
             "Reverting to `cfg.model.init_device='cpu'`.")
         cfg.model.init_device = 'cpu'
 
+    te_config = cfg.get('te_config', None)
+    te_config = om.to_container(te_config, resolve=True) if te_config else None
+    if not te_installed and te_config:
+        warnings.warn(
+                "Transformer Engine is not installed but te_config is given! "
+                "Please install transformer engine with "
+                "pip install --upgrade git+https://github.com/NVIDIA/TransformerEngine.git@stable")
+    if not te_installed and not te_config['linear_only']:
+        print('use attn_impl == Triton to take care of key_padding_mask with TransformerEngine...')
+        cfg.attn_impl = 'triton'
+
+    if te_installed and te_config and init_device == 'meta':
+        warnings.warn(
+            "Using `cfg.model.init_device='meta'` doesn't work with transformer engine! "
+            "Reverting to `cfg.model.init_device='cpu'`.")
+        cfg.model.init_device = 'cpu'
```

Comment:
open question: should we be overriding settings vs. throwing an error? e.g. forcing the init device out of meta can have other implications, and maybe forcing the user to switch it might be a better experience?